src/plugins/remote-cache/routes/get-artifact.ts

```diff
@@ -24,7 +25,7 @@ export const getArtifact: RouteOptions<
 
     try {
       const artifact = await this.location.getCachedArtifact(artifactId, teamId)
-      reply.send(artifact)
+      return reply.send(artifact)
```

Comment:
Do not return and use reply send together.

Reply:
According to fastify's docs:
https://www.fastify.io/docs/latest/Reference/Reply/#streams
> When using async-await you will need to return or await the reply object

Reply:
We should then also add that specific header. This must be changed from 3 to 4. Thanks for pointing this out.